fastlane/lib/fastlane/actions/build_app.rb

```diff
@@ -1,13 +1,160 @@
 module Fastlane
   module Actions
-    require 'fastlane/actions/build_ios_app'
-    class BuildAppAction < BuildIosAppAction
-      #####################################################
-      # @!group Documentation
-      #####################################################
+    module SharedValues
+      IPA_OUTPUT_PATH ||= :IPA_OUTPUT_PATH
+      PKG_OUTPUT_PATH ||= :PKG_OUTPUT_PATH
+      DSYM_OUTPUT_PATH ||= :DSYM_OUTPUT_PATH
+      XCODEBUILD_ARCHIVE ||= :XCODEBUILD_ARCHIVE # originally defined in XcodebuildAction
+    end
+
+    class BuildAppAction < Action
+      # rubocop:disable Metrics/PerceivedComplexity
```

Comment:
I think we can refactor this method a little bit to avoid this disable. Let's make 2 methods for the big conditional branch that accept a `export_options` parameter, and return the transformed `export_options`.

We can probably also rework this to exit early with `show_xcode_9_warning` in the correct case.

Reply:
This is no longer complex now that we reworked the ipa/pkg/dsym logic below 😊